multicast/multicast.go

```diff
@@ -0,0 +1,438 @@
+			}
+			var formattedOutput strings.Builder
+			formattedOutput.WriteString(fmt.Sprintf("%-*s\t", maxNodeNameLength+5, pod.Spec.NodeName))
+			firstLine := true
+			for _, line := range outputLines {
+				if line != "" {
+					if !firstLine {
+						formattedOutput.WriteString(fmt.Sprintf("\n%-*s\t", maxNodeNameLength+5, ""))
+					}
+					formattedOutput.WriteString(line)
+					firstLine = false
+				}
+			}
+
+			var mu sync.Mutex
+			mu.Lock()
+			fmt.Fprintln(m.params.Writer, formattedOutput.String())
+			mu.Unlock()
+		}(ciliumPod)
+	}
+
+	wg.Wait()
+	close(errCh)
+
+	var errRet error
+	for fetchdata := range errCh {
+		if fetchdata != nil {
+			errRet = errors.Join(errRet, fetchdata)
+		}
+	}
+	return errRet
+}
+
+// ListSubscriber lists multicast subscribers in every node for the specified multicast group or all multicast groups
+func (m *Multicast) ListSubscribers() error {
+	if m.params.MulticastGroupIP == "" && !m.params.All {
+		return fmt.Errorf("group-ip or all flag must be specified")
+	} else if m.params.MulticastGroupIP != "" && m.params.All {
+		return fmt.Errorf("only one of group-ip or all flag must be specified")
+	}
+
+	var target string
+	if m.params.All {
+		target = "all"
+	} else {
+		target = m.params.MulticastGroupIP
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), m.params.WaitDuration)
+	defer cancel()
+
+	ciliumPodsList, err := m.client.ListPods(ctx, m.params.CiliumNamespace, metav1.ListOptions{LabelSelector: defaults.AgentPodSelector})
+	if err != nil {
+		return err
+	}
+	ciliumPods := ciliumPodsList.Items
+
+	var wg sync.WaitGroup
+	errCh := make(chan error, len(ciliumPods))
+	wg.Add(len(ciliumPods))
+
+	maxNodeNameLength := 0
+	for _, pod := range ciliumPods {
+		if len(pod.Spec.NodeName) > maxNodeNameLength {
+			maxNodeNameLength = len(pod.Spec.NodeName)
+		}
+	}
+	if maxNodeNameLength < len("Node") {
+		maxNodeNameLength = len("Node")
+	}
+
+	//Same as cilium-dbg/cmd/multicast_subscriber.go in cilium/cilium
+	w := tabwriter.NewWriter(m.params.Writer, 16, 1, 0, ' ', 0)
+	fmt.Fprintf(w, "%-*s\t%s\n", maxNodeNameLength+5, "Node", "Group\tSubscriber\tType\t")
+	w.Flush()
+
+	for _, ciliumPod := range ciliumPods {
+		go func(pod corev1.Pod) {
+			defer wg.Done()
+			// List multicast subscribers
+			cmd := []string{"cilium-dbg", "bpf", "multicast", "subscriber", "list", target}
+			output, stdErr, err := m.client.ExecInPodWithStderr(ctx, pod.Namespace, pod.Name, defaults.AgentContainerName, cmd)
+			if err != nil {
+				if strings.Contains(stdErr.String(), "does not exist") {
+					fmt.Fprintf(m.params.Writer, "Multicast group %s does not exist in %s\n", target, pod.Spec.NodeName)
+					return
+				}
+				errCh <- err
+				return
+			}
+			outputLines := strings.Split(output.String(), "\n")
+			if len(outputLines) > 1 {
+				outputLines = outputLines[1:]
+			}
+
+			var formattedOutput strings.Builder
+			formattedOutput.WriteString(fmt.Sprintf("%-*s\t", maxNodeNameLength+5, pod.Spec.NodeName))
```

Comment:
For tabular output, we prefer to use the `tabwriter` package for alignment.

See https://github.com/cilium/cilium-cli/blob/main/bgp/routes.go#L257 for an example.